qt/python/mantidqt/project/decoderfactory.py

```diff
@@ -17,12 +19,25 @@ def find_decoder(cls, tag):
         """
         This assumes that tag is present in a decoder and if it cannot find a decoder with the tag it raises ValueError
         :param tag: String; The tag used for saving and thus to be reused for loading that encoder back.
-        :return: Decoder object; The object of a decoder for the tag that was passed.
+        :return: Decoder object; The object of a decoder for the tag that was passed. Or None if none were found.
         """
         for decoder in cls.decoder_list:
             if tag in decoder.tags:
                 return decoder()
-        raise ValueError("Unable to find decoder for tag: " + tag + " Please add the decoder for: " + tag)
+        return None
+
+    @classmethod
+    def decode(cls, tag, dic, directory=None):
+        decoder = cls.find_decoder(tag)
```

Comment:
Can we have some documentation for the method and parameters, please?

Similarly for a few of the other new functions added in this change set.